common/src/lib.rs

```diff
@@ -215,31 +318,26 @@ pub fn get_code_metadata(code_hash: H256) -> Option<CodeMetadata> {
         .map(|data| CodeMetadata::decode(&mut &data[..]).expect("data encoded correctly"))
 }
 
-/// Enumeration contains variants for program state.
-#[derive(Clone, Debug, Decode, Encode, PartialEq, TypeInfo)]
-pub enum ProgramState {
-    /// `init` method of a program has not yet finished its execution so
-    /// the program is not considered as initialized. All messages to such a
-    /// program go to the wait list.
-    /// `message_id` contains identifier of the initialization message.
-    Uninitialized { message_id: H256 },
-    /// Program has been successfully initialized and can process messages.
-    Initialized,
-}
-
-pub fn get_program_state(id: H256) -> Option<ProgramState> {
-    get_program(id).map(|p| p.state)
-}
-
 pub fn set_program_initialized(id: H256) {
-    if let Some(mut p) = get_program(id) {
+    if let Some(Active(mut p)) = get_program(id) {
```

Comment:
Should we provide panic or return `Err(_)` for not-found cases etc?

Reply:
Yes, will introduce panic. To consider: we discussed using strong types in pallet (and common crate as well), which will prevent such cases (inappropriate use), check storage transition principles.

Reply:
Agreed, assertions (`expect`) should be on the calling side. When remove common (move to pallet) we need to rewrite it with strong types to avoid wrong storage state transitions, for example, setting program initialized for terminated programs.

Reply:
Could you, please, open the issue for this in `Gear Agenda` project and resolve this conversation

Reply:
https://github.com/gear-tech/gear/issues/660